libs/vectorise/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ setup_compiler()
 # ------------------------------------------------------------------------------
 
 setup_library(fetch-vectorise)
-target_link_libraries(fetch-vectorise INTERFACE fetch-meta pthread)
+target_link_libraries(fetch-vectorise INTERFACE fetch-meta pthread fetch-core)
```

Comment:
this looks like a circular dependency!!!!
core depends on vectorise right?
we should not permit this